File: backend/src/main/java/com/example/exammanagement/service/ExamService.java
package com.example.exammanagement.service;

import com.example.exammanagement.entity.Exam;
import com.example.exammanagement.repository.ExamRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ExamService {
    
    @Autowired
    private ExamRepository examRepository;
    
    public List<Exam> getAllExams() {
        return examRepository.findAll();
    }
    
    public Exam getExamById(Long id) {
        return examRepository.findById(id)
            .orElseThrow(() -> new RuntimeException("Exam not found with id: " + id));
    }
    
    public List<Exam> getExamsByCourse(Long courseId) {
        return examRepository.findByCourseId(courseId);
    }
    
    public Exam createExam(Exam exam) {
        return examRepository.save(exam);
    }
    
    public Exam updateExam(Long id, Exam examDetails) {
        Exam exam = getExamById(id);
        
        exam.setTitle(examDetails.getTitle());
        exam.setDescription(examDetails.getDescription());
        exam.setExamDate(examDetails.getExamDate());
        exam.setDuration(examDetails.getDuration());
        exam.setTotalMarks(examDetails.getTotalMarks());
        exam.setCourse(examDetails.getCourse());
        
        return examRepository.save(exam);
    }
    
    public void deleteExam(Long id) {
        Exam exam = getExamById(id);
        examRepository.delete(exam);
    }
}

File: frontend/src/services/examService.js
import api from './api'

export const examService = {
  async getExams() {
    const response = await api.get('/exams')
    return response.data
  },

  async getExamById(id) {
    const response = await api.get(`/exams/${id}`)
    return response.data
  },

  async createExam(examData) {
    const response = await api.post('/exams', examData)
    return response.data
  },

  async updateExam(id, examData) {
    const response = await api.put(`/exams/${id}`, examData)
    return response.data
  },

  async deleteExam(id) {
    const response = await api.delete(`/exams/${id}`)
    return response.data
  },

  async getExamsByCourse(courseId) {
    const response = await api.get(`/exams/course/${courseId}`)
    return response.data
  },

  async getExamsByFaculty(facultyId) {
    const response = await api.get(`/exams/faculty/${facultyId}`)
    return response.data
  },
}
